game/src/story/vwf.asm

```diff
@@ -0,0 +1,743 @@
+SECTION "VWF Variables", WRAMX[$DC00], BANK[$1]
+VWFTilesDrawn:: ds 1
+VWFCurrentLetter:: ds 1
+VWFControlCodeArguments:: ds 2
+VWFCompositeAreaAddress:: ds 2
+VWFLetterShift:: ds 1
+VWFOldTileMode:: ds 1
+VWFTileBaseIdx:: ds 1
+VWFIsInit:: ds 1
+VWFIsSecondLine:: ds 1
+VWFTrackBank:: ds 1
+
+SECTION "VWF Composite Area", WRAMX[$DCD0], BANK[$1]
+VWFCompositeArea:: ds $30
+
+SECTION "VWF Drawing Functions", ROMX[$6000], BANK[$24]
+VWFDrawLetterTable::
+	; This determines the width of each character (excluding the 1px between characters).
+	; The address of this table must be a multiple of $100.
+
+	;  x0 x1 x2 x3 x4 x5 x6 x7 x8 x9 xA xB xC xD xE xF
+	db 2, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; 0x
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; 1x
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; 2x
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; 3x
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; 4x
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; 5x
+	db 7, 7, 4, 4, 4, 3, 1, 3, 4, 5, 5, 4, 4, 4, 4, 4 ; 6x
+	db 4, 4, 4, 4, 4, 5, 1, 5, 4, 5, 4, 7, 7, 1, 4, 7 ; 7x
+	db 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5 ; 8x
+	db 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 4, 4, 4, 4, 4 ; 9x
+	db 4, 4, 1, 2, 4, 1, 5, 4, 4, 4, 4, 4, 4, 4, 4, 4 ; Ax
+	db 5, 4, 4, 4, 3, 5, 5, 5, 2, 2, 4, 1, 7, 1, 7, 2 ; Bx
+	db 2, 3, 5, 3, 4, 2, 6, 5, 5, 7, 7, 7, 7, 4, 7, 2 ; Cx
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; Dx
+	db 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7 ; Ex
+	db 7, 7, 3, 7, 3, 3, 3, 3, 4, 4, 7, 5, 7, 7, 7, 7 ; Fx
+	;  x0 x1 x2 x3 x4 x5 x6 x7 x8 x9 xA xB xC xD xE xF
+
+VWFFont::
+	INCBIN "build/tilesets/Font.vwffont"
+
+VWFMessageBoxInputHandler::
+	; Advance on button press.
+
+	ld a, [hJPInputChanged]
+	and hJPInputA + hJPInputB
+	ret nz
+
+	; Auto-advance if button held down.
+
+	ld a, [hJPInputHeldDown]
+	and hJPInputA + hJPInputB
+	ret z
+
+	; Wait 10 frames before advancing.
+
+	ld a, [$c6c5]
+	cp $10
+	jr nz, .buttonNotPressedLongEnough
+	xor a
+	inc a
+	ret
+
+.buttonNotPressedLongEnough
+	inc a
+	ld [$c6c5], a
+	xor a
+	ret
+
+.buttonPressed
+
+VWFIncTextOffset::
```

Comment:
If you move this to patch/, you can just make the existing functions into globals and use those instead.